### PR TITLE
feat: migrate hero banner - FRONT-1771

### PR DESCRIPTION
**Pull request by kalinchernev**

### CODE REVIEW

File: src/implementations/twig/components/hero-banner/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- **"description"** (string) (default: '') Description of banner
+- **"centered"** (bool) (default: true) Define if banner should be centered
+- [DEPRECATED] **"button"** (associative array) (default: predefined structure) predefined structure for EC Button component
+- **"link"** (associative array) (default: predefined structure) predefined structure for EC Link component
+- **"extra_classes"** (optional) (string) (default: '') Extra classes (space separated)
+- **"extra_attributes"** (optional) (array) (default: []) Extra attributes
+  - "name" (string) Attribute name, eg. 'data-test'
+  - "value" (string) Attribute value, eg: 'data-test-1',
+- **"_compliance_"**: (optional) (boolean) (default: false) Activates debug
+- **"_compliance_inner_check_"** (optional) (boolean) (default: false) Inline compliance report
```
Comment (planctus) [inline src/implementations/twig/components/hero-banner/README.md]:
```suggestion
- **"\_compliance_inner_check\_"** (optional) (boolean) (default: false) Inline compliance report
```

File: src/implementations/twig/components/hero-banner/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- **"image"** (string) (default: '') Image for banner (required for image banner type)
+- **"description"** (string) (default: '') Description of banner
+- **"centered"** (bool) (default: true) Define if banner should be centered
+- [DEPRECATED] **"button"** (associative array) (default: predefined structure) predefined structure for EC Button component
+- **"link"** (associative array) (default: predefined structure) predefined structure for EC Link component
+- **"extra_classes"** (optional) (string) (default: '') Extra classes (space separated)
+- **"extra_attributes"** (optional) (array) (default: []) Extra attributes
+  - "name" (string) Attribute name, eg. 'data-test'
+  - "value" (string) Attribute value, eg: 'data-test-1',
+- **"_compliance_"**: (optional) (boolean) (default: false) Activates debug
```
Comment (planctus) [inline src/implementations/twig/components/hero-banner/README.md]:
```suggestion
- **"\_compliance\_"**: (optional) (boolean) (default: false) Activates debug
```

File: src/presets/dev/src/dev-print.scss [hunk 3]
```diff
@@ -11,6 +11,7 @@
 @use '@ecl/vanilla-component-expandable/expandable-print';
 @use '@ecl/vanilla-component-fact-figures/fact-figures-print';
 @use '@ecl/vanilla-component-file/file-print';
+@use '@ecl/vanilla-component-hero-banner/hero-banner-print';
```
Comment (planctus) [inline src/presets/dev/src/dev-print.scss]: shouldn't come after form-print, alphabetically..?

File: src/presets/dev/src/dev.scss [hunk 4]
```diff
@@ -14,6 +14,7 @@
 @use '@ecl/vanilla-component-expandable/expandable';
 @use '@ecl/vanilla-component-fact-figures/fact-figures';
 @use '@ecl/vanilla-component-file/file';
+@use '@ecl/vanilla-component-hero-banner/hero-banner';
```
Comment (planctus) [inline src/presets/dev/src/dev.scss]: and this after form?
Alphabetically yes, but the issue here is that we get the `.ecl-hero-banner__link-cta` value for the margin-top overriden by the `.ecl-link`.
image

File: src/specs/components/hero-banner/demo/data--image-box.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  title: 'EU Budget for the future',
+  description: 'Innovation, economy, environment and geopolitics',
+  link: {
+    link: {
+      label: 'Subscribe',
+      path: '/example',
+      aria_label: '',
```
Comment (planctus) [inline src/specs/components/hero-banner/demo/data--image-box.js]: why an empty prop in the specs?

File: src/specs/components/hero-banner/demo/data--image-gradient.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  title: 'EU Budget for the future',
+  description: 'Innovation, economy, environment and geopolitics',
+  link: {
+    link: {
+      label: 'Subscribe',
+      path: '/example',
+      aria_label: '',
```
Comment (planctus) [inline src/specs/components/hero-banner/demo/data--image-gradient.js]: same empty prop in all the specs file, better removing it if there is no required aria_label attr

File: src/implementations/twig/components/hero-banner/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+### Parameters
+
+- **"type"** (string) (default: 'default') Type of banner (can be 'default','image','image-shade','primary')
+- **"title"** (string) (default: '') Title of banner
+- **"image"** (string) (default: '') Image for banner (required for image banner type)
+- **"description"** (string) (default: '') Description of banner
+- **"centered"** (bool) (default: true) Define if banner should be centered
+- [DEPRECATED] **"button"** (associative array) (default: predefined structure) predefined structure for EC Button component
+- **"link"** (associative array) (default: predefined structure) predefined structure for EC Link component
```
Comment (planctus) [inline src/implementations/twig/components/hero-banner/README.md]: Since we are removing references to the system let's also remove from it here
```suggestion
- **"link"** (associative array) (default: predefined structure) predefined structure for the Link component
```

File: src/implementations/twig/components/hero-banner/README.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+### Parameters
+
+- **"type"** (string) (default: 'default') Type of banner (can be 'default','image','image-shade','primary')
+- **"title"** (string) (default: '') Title of banner
+- **"image"** (string) (default: '') Image for banner (required for image banner type)
+- **"description"** (string) (default: '') Description of banner
+- **"centered"** (bool) (default: true) Define if banner should be centered
+- [DEPRECATED] **"button"** (associative array) (default: predefined structure) predefined structure for EC Button component
```
Comment (planctus) [inline src/implementations/twig/components/hero-banner/README.md]:
```suggestion
- [DEPRECATED] **"button"** (associative array) (default: predefined structure) predefined structure for the Button component
```

File: src/implementations/twig/components/hero-banner/README.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+npm package: `@ecl/twig-component-hero-banner`
+
+```shell
+npm install --save @ecl/twig-component-hero-banner
+```
+
+### Parameters
+
+- **"type"** (string) (default: 'default') Type of banner (can be 'default','image','image-shade','primary')
```
Comment (planctus) [inline src/implementations/twig/components/hero-banner/README.md]: types seems to be now `white` `grey` `image` `image-gradient` `primary` `image-shade`, this was not kept up to date in ecl-twig..

File: src/implementations/twig/components/hero-banner/hero-banner.html.twig [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+{#
+  Parameters:
+    - "type" (string) (default: 'primary') Type of banner (can be 'grey','white', 'image','image-shade','image-gradient')
+    - "title" (string) (default: '') Title of banner
+    - "image" (string) (default: '') Image for banner (required for image banner type)
+    - "description" (string) (default: '') Description of banner
+    - "centered" (bool) (default: true) Define if banner should be centered
+    - [DEPRECATED] "button" (associative array) (default: predefined structure) predefined structure for EC Button component
+    - "link" (associative array) (default: predefined structure) predefined structure for EC Link component
```
Comment (planctus) [inline src/implementations/twig/components/hero-banner/hero-banner.html.twig]:
```suggestion
    - "link" (associative array) (default: predefined structure) predefined structure for the Link component
```

File: src/implementations/twig/components/hero-banner/hero-banner.html.twig [hunk 11]
```diff
@@ -0,0 +1,108 @@
+{% spaceless %}
+
+{#
+  Parameters:
+    - "type" (string) (default: 'primary') Type of banner (can be 'grey','white', 'image','image-shade','image-gradient')
+    - "title" (string) (default: '') Title of banner
+    - "image" (string) (default: '') Image for banner (required for image banner type)
+    - "description" (string) (default: '') Description of banner
+    - "centered" (bool) (default: true) Define if banner should be centered
+    - [DEPRECATED] "button" (associative array) (default: predefined structure) predefined structure for EC Button component
```
Comment (planctus) [inline src/implementations/twig/components/hero-banner/hero-banner.html.twig]:
```suggestion
    - [DEPRECATED] "button" (associative array) (default: predefined structure) predefined structure for the Button component
```

**Review by planctus — CHANGES_REQUESTED** [review]

File: src/specs/components/hero-banner/demo/data--image-box.js (hunk 5 above)
Comment (kalinchernev) [reply src/specs/components/hero-banner/demo/data--image-box.js]: I took the rendered data from ecl-twig 1:1

File: src/specs/components/hero-banner/demo/data--image-gradient.js (hunk 6 above)
Comment (kalinchernev) [reply src/specs/components/hero-banner/demo/data--image-gradient.js]: What I did and have previously done for the other components is to start the ecl-twig storybook instance and put a breakpoint at the point where data is prepared and passed to the twig template. It's easier than following along logic in adapters and sample data reformatting.

File: src/implementations/twig/components/hero-banner/README.md (hunk 9 above)
Comment (kalinchernev) [reply src/implementations/twig/components/hero-banner/README.md]: I'm not sure I understand your feedback here. I've taken the README file from the ecl-twig repo as well as its corresponding twig template file. If this line is somehow not matching your expectations, this probably means ecl-twig implementation should be updated in the other repo if we are to change something here.

File: src/presets/dev/src/dev.scss (hunk 4 above)
Comment (kalinchernev) [reply src/presets/dev/src/dev.scss]: Noob mistake, sorry. I looked at https://github.com/ec-europa/europa-component-library/blob/v2-dev/src/systems/ec/implementations/vanilla/packages/ec-preset-dev/src/ec-preset-dev-print.scss and updated both print and normal stylesheets correspondingly.

File: src/specs/components/hero-banner/demo/data--image-box.js (hunk 5 above)
Comment (planctus) [reply src/specs/components/hero-banner/demo/data--image-box.js]: Yeah, i see, but that doesn't make it right, if you think it is, fine, but i struggle to understand this logic.

File: src/implementations/twig/components/hero-banner/README.md (hunk 9 above)
Comment (planctus) [reply src/implementations/twig/components/hero-banner/README.md]: My expectations? These are the "types" that the component supports, not really a preference of mine. If you don't want to fix this here let's merge it like this, but it's wrong and it's hard to follow this kind of reasoning instead of simply doing the right thing.

**Review by planctus — APPROVED** [review]

File: src/implementations/twig/components/hero-banner/README.md (hunk 9 above)
Comment (kalinchernev) [reply src/implementations/twig/components/hero-banner/README.md]: I like the the "doing the right thing", thank you for keeping this spirit.
I updated the files which I hope is done right.

File: src/specs/components/hero-banner/demo/data--image-box.js (hunk 5 above)
Comment (kalinchernev) [reply src/specs/components/hero-banner/demo/data--image-box.js]: Still doing the right thing, I removed the empty items from spec files. There is no logic here, copy-paste of the code created in ecl-twig repository through all the data files there.

**Comment by kalinchernev** [comment]
Requested another round of review after changes

**Comment by planctus** [comment]
There are conflicts, @kalinchernev, you solve them and we merge this? ;)
